indico/modules/events/contributions/models/contributions.py

```diff
@@ -332,6 +332,21 @@ def __table_args__(cls):
         )
     )
 
+    @declared_attr
+    def paper_last_revision(cls):
+        # Incompatible with joinedload
+        return db.relationship(
+            'PaperRevision',
+            uselist=False,
+            lazy=True,
+            viewonly=True,
+            primaryjoin=(db.and_((PaperRevision.contribution_id == cls.id),
+                                 (PaperRevision.submitted_dt == (db.select([db.func.max(PaperRevision.submitted_dt)])
```

Comment:
I would assign the subquery to a `subquery` variable and use `PaperRevision.submitted_dt == subquery.as_scalar()` here (or even move the `as_scalar()` to the `subquery = ...` too)